### DISCUSSION TITLE
[ERROR] An error occurred (InvalidIPAddress.InUse) when calling the RunInstances operation: Address XXX.XXX.XXX.XXX is in use.

### DISCUSSION THREAD

**Discussion by juanhurtado00** [aws/aws-cli#7176] [Q&A] [answered]
I HAVE AN AUTOMATION THROUGH LAMBDAS FOR THE CREATION OF EC2, IT IS GENERATING ME THE FOLLOWING ERROR:

[ERROR] An error occurred (InvalidIPAddress.InUse) when calling the RunInstances operation: Address 10.xxx.xxx.xxx is in use.

THE CODE IS THE FOLLOWING IN PYTON 3.9:

`
import os
import boto3
from dateutil import tz
import datetime
from botocore.exceptions import ClientError

KEY = 'REGION'
ENV = 'ENV'
AWS_ENV = os.getenv(ENV)
AWS_REGION = os.getenv(KEY)
AWS_SESSION = boto3.Session(region_name=AWS_REGION)

from_zone = tz.gettz('UTC')
to_zone = tz.gettz('America/Bogota')
now = datetime.datetime.now()
now = now.replace(tzinfo=from_zone)

dateFormat = now.astimezone(to_zone).strftime("%X")
date = now.astimezone(to_zone).strftime("%Y%m%d")
ec2_resource = AWS_SESSION.resource('ec2')
ec2_client = AWS_SESSION.client('ec2')
vpc_resource = ec2_resource.Vpc('Id')

def get_ami():

    id = []

    try:
        images = ec2_resource.images.filter(
            Filters=[
                {
                    'Name': 'tag:Env',
                    'Values': [AWS_ENV]
                }
            ]
        )

        for i in images:
            id.append(i)

        return id

    except:
        return None

def get_resources():
    
    return {
            'IP-SERVER-NAME-DEV': '10.XXX.XXX.XXX',
            'TYPE-SERVER-NAME-DEV': 't3a.medium',
            'IP-SERVER-NAME-DEV': '10.XXX.XXX.XXX',
            'TYPE-SERVER-NAME-DEV': 't3a.medium',
            'IP-SERVER-NAME-DEV': '10.XXX.XXX.XXX',
            'TYPE-SERVER-NAME-DEV': 't3a.medium',
            }

def create_ec2(id_ami,instance_type,sg_id,subnet_id,private_ip,ec2_name):
    
    try:
        ec2_resource.create_instances(
            BlockDeviceMappings=[
                {
                    'DeviceName': '/dev/sda1',
                    'Ebs': {
                        'DeleteOnTermination': True,
                        'VolumeType': 'gp3'
                    },
                },
            ],
            ImageId=id_ami.id,
            InstanceType=instance_type,
            MaxCount=1,
            MinCount=1,
            Monitoring={
                'Enabled': False
            },
            SecurityGroupIds=[sg_id],
            SubnetId=subnet_id,
            PrivateIpAddress=private_ip,
            TagSpecifications=[
                {
                    'ResourceType': 'instance',
                    'Tags': [
                        {
                            'Key': 'Name',
                            'Value': 'EC2-'+ec2_name
                        },
                        {
                            'Key': 'Env',
                            'Value': AWS_ENV
                        }
                    ]
                },
                {
                    'ResourceType': 'volume',
                    'Tags': [
                        {
                            'Key': 'Name',
                            'Value': 'EBS-'+ec2_name
                        },
                        {
                            'Key': 'Env',
                            'Value': AWS_ENV
                        }
                    ]
                },
                {
                    'ResourceType': 'network-interface',
                    'Tags': [
                        {
                            'Key': 'Name',
                            'Value': 'ENI-'+ec2_name
                        },
                        {
                            'Key': 'Env',
                            'Value': AWS_ENV
                        }
                    ]
                }
            ]
        )

        return True

    except ClientError as e:
        mensaje = dateFormat+' [ERROR] '+str(e)
        print(mensaje)
       
--------------------------------------------------------------------------------------------------------

THE LAMBDA FUNCTION WAS WORKING WELL 2 WEEKS AGO, BUT SINCE THEN IT IS GENERATING THE ERROR WHEN IT IS RUN, IT CREATES THE EC2 WITH ALL THE ATTRIBUTES, BUT IT SEND ERROR.

PLEASE YOUR HELP VALIDATING IF THE CODE IS WRONG OR YOU HAVE TO GENERATE A CHANGE

**Comment by tim-finnigan** [comment, marked as answer]
Hi @juanhurtado00, it looks like this issue relates to boto3 rather than the AWS CLI, in which case the appropriate place for this is here: https://github.com/boto/boto3/discussions

But I can refer you to this EC2 documentation for the `InvalidIPAddress.InUse` error: https://docs.aws.amazon.com/AWSEC2/latest/APIReference/errors-overview.html

> The specified IP address is already in use. If you are trying to release an address, you must first disassociate it from the instance.
